subworkflows/nf-core/fasta_newick_epang_gappa/main.nf

```diff
@@ -0,0 +1,156 @@
+include { HMMER_HMMBUILD as HMMER_HMMBUILD          } from '../../../modules/nf-core/hmmer/hmmbuild/main'
+include { HMMER_HMMALIGN as HMMER_HMMALIGNREF       } from '../../../modules/nf-core/hmmer/hmmalign/main'
+include { HMMER_HMMALIGN as HMMER_HMMALIGNQUERY     } from '../../../modules/nf-core/hmmer/hmmalign/main'
+include { HMMER_ESLALIMASK as HMMER_MASKREF         } from '../../../modules/nf-core/hmmer/eslalimask/main'
+include { HMMER_ESLALIMASK as HMMER_MASKQUERY       } from '../../../modules/nf-core/hmmer/eslalimask/main'
+include { HMMER_ESLREFORMAT as HMMER_UNALIGNREF     } from '../../../modules/nf-core/hmmer/eslreformat/main'
+include { HMMER_ESLREFORMAT as HMMER_AFAFORMATREF   } from '../../../modules/nf-core/hmmer/eslreformat/main'
+include { HMMER_ESLREFORMAT as HMMER_AFAFORMATQUERY } from '../../../modules/nf-core/hmmer/eslreformat/main'
+include { MAFFT                                     } from '../../../modules/nf-core/mafft/main'
+include { EPANG_PLACE                               } from '../../../modules/nf-core/epang/place/main'
+include { EPANG_SPLIT                               } from '../../../modules/nf-core/epang/split/main'
+include { GAPPA_EXAMINEGRAFT as GAPPA_GRAFT         } from '../../../modules/nf-core/gappa/examinegraft/main'
+include { GAPPA_EXAMINEASSIGN as GAPPA_ASSIGN       } from '../../../modules/nf-core/gappa/examineassign/main'
+include { GAPPA_EXAMINEHEATTREE as GAPPA_HEATTREE   } from '../../../modules/nf-core/gappa/examineheattree/main'
+
+workflow FASTA_NEWICK_EPANG_GAPPA {
+
+    take:
+    ch_pp_data // channel: [ meta: val(meta), data: [ alignmethod: val(alignmethod), queryseqfile: file(queryseqfile), refseqfile: file(refseqfile), refphylogeny: file(refphylogeny), hmmfile: file(hmmfile), model: val(model) ] ]
```

Comment:
It's preffered to split this channel into multiple input channels that will get merged in the subworkflow. This reduces the complexity for the user too (so you don't have to use arrays in arrays etc)

Reply:
The problem is that that's difficult to synchronise if the pieces come from different sources, i.e. to make sure that they come in the same order. (At least I don't know how to do that.)

Reply:
You can join the channels on meta. Just gotta make sure that all channels that have to be merged do contain exactly the same meta. (You can use the `.join()` operator for this: https://www.nextflow.io/docs/latest/operator.html#join)

Reply:
Right. I'd like to give some more context. The workflow is called by my phyloplace pipeline (it's basically the whole pipeline). The sample sheet for the pipeline, parsed with `splitCsv()`, is turned into a single channel with this structure. If I were to give the subworkflow separate channels, I'd have to split the channel before calling the subworkflow, then `join()` them in the subworkflow. 

I chose the map structure for the channel because I personally prefer to have named inputs when the number of inputs is large.

I plan to include the subworkflow also into ampliseq. I don't foresee any problems calling it with this structure. (Of course it should be documented though. That was an oversight of mine mostly because I'm not so familiar with the documentation format.)

Moreover, I was recently advised to merge channels into one when submitting a module. An advise that I later was grateful for, because I found it more explicit to `join()` before calling the module rather than the module doing that for me.

Reply:
While I agree that the input channel here is a little more complex than for other workflows, I find it quite intuitive due to the naming. I am fine with it as is. @nvnieuwk is there a rule for input channels?

Reply:
Aaah okay the context is very clear, thanks! You can leave it as it is :)